docs/Creating-Your-Own-EmulationStation-Theme.md

```diff
@@ -1431,31 +1455,199 @@ And my complete `theme.xml`, which is identical in each system folder:
 </theme>
 ```
 
----
+## Tips and Tricks
+
+### Keyboard Shortcuts in EmulationStation
+
+When you start EmulationStation with the debug flag on the command line, you have these keyboard combos available (to be used with _left_ ++ctrl++):
+
+- ++ctrl+r++: Reload all UI views (system carousel and in-system view)
+- ++ctrl+i++: Toggle visibility the image components (`<image ...>`) boundary boxes
+- ++ctrl+t++: Toggle visibility the text component (`<text ...>`) boundary boxes
+- ++ctrl+g++: Toggle visibility Gridlayout boundary boxes (if you develop a theme for grid layout)
+
+### The Size Element for Text Components
+
+The `<size/>` element takes a pair of values. However, depending on the values of the pair you may also control the layout of the text element. This section will illustrate the different values and their impact on the layout. 
+
+As you may have already noticed the size element takes percent values of the actual screen size for _images_. For the layout of _text components_ with a given <size/>, additionally the selected font (its [Glyphs](https://en.wikipedia.org/wiki/Glyph)), the `<fontSize/>` and the defined `<lineSpacing/>` are factors for the layout. The last two values are also expected to be percent values. The glyph values are calculated by ES depending on the `<font/>` the theme has defined.
+
+The pair of values is referenced as `width` (x-value) and `height` (y-value) e.g., `<size>width height</size>`.
+
+The actual height of one line of text (`lineHeight`) is calculated by the `height` times the `lineSpacing` [times the largest height of the glyphs](https://github.com/RetroPie/EmulationStation/blob/9afa234f3c779ca23f832141f819886a984c1ee6/es-core/src/resources/Font.cpp#L476) times `fontSize`.
+
+For your information: The images are taken from the [Epic Noir](https://github.com/c64-dev/es-theme-epicnoir) theme on a 5:4 display (1280x1024), cropped for this tutorial.
+
+#### Case 1: Overly Large Size
+
+<!-- case 1: x>0, y > one lineheight -->
+<figure markdown>
+  ![](images/es-themes/th_01_epnoir_asis.png)
+  <figcaption>Case 1: Text bounding box <code>height</code> greater than wrapped text height</figcaption>
+</figure>
+
+This shows the theme layout as-is when you check it out from GitHub. Notice how the "Games Available" bounding box (bbox) and the bbox of the text below overlap (the latter is named long description or `longdescription`). However, the texts do not overlap coincidentally as the bbox for the long description is large enough, and the text is put in the middle of the text bbox by default (instead of top or bottom vertical alignment).
+
+This is the respective section in the theme file. Note line #8 and #21 (using same position) and #22 (size of long description):
```

Comment:
The lines can be highlighted by adding `hl_lines=8 21 22` to the XML codeblock header.

Reply:
Thanks. Done.